storage-proofs-porep/src/stacked/vanilla/proof_ext.rs

```diff
@@ -0,0 +1,67 @@
+use scheduler_client::{
+    register, schedule_one_of, ResourceAlloc, ResourceMemory, ResourceReq, ResourceType, TaskFunc,
+    TaskReqBuilder, TaskResult, TaskType,
+};
+use std::time::Duration;
+use storage_proofs_core::error::{Error, Result};
+
+// TODO: We need to define if this is going to be part of the configuration file
+// as well as an appropiate value for this.
+const TIMEOUT: u64 = 3600;
+
+pub(crate) struct Builder<F>
+where
+    for<'a> F: FnMut(Option<&'a ResourceAlloc>) -> Result<TaskResult, Error>,
+{
+    call: F,
+    num_iter: usize,
+}
+
+impl<F> Builder<F>
+where
+    for<'a> F: FnMut(Option<&'a ResourceAlloc>) -> Result<TaskResult, Error>,
+{
+    pub(crate) fn new(call: F, num_iter: usize) -> Self {
+        Self { call, num_iter }
+    }
+
+    pub(crate) fn build(&mut self) -> Result<(), Error> {
+        let requirements = {
+            let resouce_req = ResourceReq {
+                resource: ResourceType::Gpu(ResourceMemory::All),
+                quantity: 1,
+                preemptible: self.num_iter > 1,
+            };
+            let task_req = TaskReqBuilder::new()
+                .with_task_type(TaskType::MerkleProof)
```

Comment:
Assuming this type is meant to describe Merkle tree building, it is misnamed. Merkle proofs are later generated over these trees, but the act of building them shouldn't reference 'proof', since that is confusing.

Reply:
I am open to set a more appropriate name here. also not sure if it should include the word __Merkle__ as what I have seen is that there are operations that not only use neptune but bellperson.